wagtail/wagtailforms/templates/wagtailforms/index_submissions.html

```diff
@@ -23,11 +23,72 @@
                 },
                 lang: 'lang'
             });
+
+            var selectAllCheckbox = document.getElementById('select-all');
+            var deleteButton = document.getElementById('delete-submissions');
+            var selectedSubmissions = {};
```

Comment:
A slightly nitpicky issue: if you use the back button (which the confirm-delete page will tend to encourage, since there's no 'cancel' button there) the browser will restore the previous state of the checkboxes. These will then be out of sync with `selectedSubmissions` and the visibility of the 'delete' button.

I think you could probably eliminate `selectedSubmissions`, and use jQuery selectors directly to set `everySubmissionSelected` / `someSubmissionsSelected`:

    var submissionCheckboxes = $('input[type=checkbox].select-submission');
    var someSubmissionsSelected = submissionCheckboxes.is(':checked');
    var everySubmissionSelected = !submissionCheckboxes.is(':not(:checked)');

Then calling `updateActions` on page load should bring everything back in sync.